src/sentry/models/auditlogentry.py

```diff
@@ -92,7 +95,10 @@ def as_event(self) -> AuditLogEvent:
         Serializes a potential audit log database entry as a hybrid cloud event that should be deserialized and
         loaded via `from_event` as faithfully as possible.
         """
-        self._apply_actor_label()
+        # TODO(hc): Denormalize user name into organization member objects so that we can save a cross silo call and
+        # also remove the following line.
+        with django_test_transaction_water_mark(db.router.db_for_read(RegionOutbox)):
+            self._apply_actor_label()
```

Comment:
An alternative would be to stop storing username/apikey data in audit log messages, and instead only store the ids. We could resolve the identifiers when the audit log payload is processed by control silo, or do it even later when rendering audit logs into API responses.

Reply:
Good call, I do like the id of resolving this when rendering.  It makes more sense in a lot of sense.  I'll see if I can make that change in a simple way.